lib/osnetwork.rb

```diff
@@ -4,55 +4,84 @@
 require_relative 'osproject_android'
 
 class NetworkHandler
-    @instance = new
+  @instance = new
 
-    private_class_method :new
+  private_class_method :new
 
-    def self.instance
-      @instance
-    end
+  URL = 'https://api.onesignal.com/api/v1/track'
 
-    def track_uri
-      URI.parse('https://api.onesignal.com/api/v1/track')
-    end
+  def self.instance
+    @instance
+  end
 
-    def getHttp_net(uri)
-      http = Net::HTTP.new(uri.host, uri.port)
-      http.use_ssl = true
-      return http
-    end
+  def get_http_net()
+    uri = URI.parse(URL)
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    return http
+  end
+
+  def send_track_from_error(app_id, platform, lang, success_mesage, error_message)
```

Comment:
What do you think of only using this function for the cases when `success_message` is `nil`?
That way you don't have to check for the `success_message` and you just handle the error case.
For the cases when `success_message` is not `nil`, you can call `send_track_command_actions` directly. Let me know if this is not clear enough.